v2/internal/test/ca.go

```diff
@@ -76,6 +79,41 @@ func (ca *CA) Bundle(td spiffeid.TrustDomain) *x509bundle.Bundle {
 	return bundle
 }
 
+func (ca *CA) PublicKey() crypto.PublicKey {
+	return ca.key.Public()
+}
+
+func (ca *CA) CreateJWTSVID(spiffeID string, audience []string) string {
+	issuer := "CA"
+	d, _ := time.ParseDuration("1h")
+	expiry := time.Now().Add(d)
+	claims := jwt.Claims{
+		Subject:  spiffeID,
+		Issuer:   issuer,
+		Audience: audience,
+		IssuedAt: jwt.NewNumericDate(time.Now()),
+		Expiry:   jwt.NewNumericDate(expiry),
+	}
+
+	jwtSigner, err := jose.NewSigner(
+		jose.SigningKey{
+			Algorithm: jose.ES256,
+			Key:       cryptosigner.Opaque(ca.key),
```

Comment:
JWT-SVIDs require the key ID. Maybe the CA should generate a new one in `NewCA`?

Here's what we do in SPIRE:
https://github.com/spiffe/spire/blob/86aa56be86535d1e069f731d635474f1a0f1ea07/pkg/server/ca/manager.go#L1110

Reply:
Excellent, thanks for that link!